### PR TITLE
Added appliance definitions for various Mikrotik hardware models simulated through the use of the Mikrotik CHR platform.

### PR DESCRIPTION
**Pull request by AzorianMatt**
I have created 10 Mikrotik appliances based on the CHR image that will simulate various router and switch models as it pertains to CPU count, RAM, and network interfaces.

Unfortunately, CHR does not pickup these interface names so one must manually update the interface names in the OS after first run. This is because CHR is designed to give standard naming to it's virtual Ethernet interfaces.

If I knew how to create a standard image from their other models then I could script a method for building them from the NPK formatted updates. In that scenario, QEMU emulation could be leveraged to create a more accurate simulation of the hardware.

---
When updating an **existing** appliance:
- [ ] The new version is on top.
- [ ] The filenames in the "images" section are unique, to avoid appliances / version overwriting each other.
- [ ] If you forked the repo, running check.py doesn't drop any errors for the updated file.
---
When creating a **new** appliance:
- It's tested locally, i.e.
  - [x] You dragged an instance into a project on your box, got it installed (if necessary), and did some basic network checks (ping, UI reachable, etc.).
  - [X] GNS3 VM can run it without any tweaks.
  - [X] The device is in the right category: router, switch, guest (hosts), firewall
  - [X] You filled in as much info as possible (checks the schemas and other appliance files for some guidance).
- [ ] When adding a container: it builds on Docker Hub and can be pulled.
- [X] The filenames in the "images" section are unique (to avoid appliances and/or versions overwriting each other).
- [X] If you forked the repo, running check.py doesn't drop any errors for the new file.
- [ ] *Optional: a symbol has been created for the new appliance.*

### CODE REVIEW

File: appliances/mikrotik-ccr1036-8g-2s+.gns3a [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+            "version": "7.1.5",
+            "md5sum": "9c0be05f891df2b1400bdae5e719898e",
+            "filesize": 134217728,
+            "download_url": "http://www.mikrotik.com/download",
+            "direct_download_url": "https://download.mikrotik.com/routeros/7.1.5/chr-7.1.5.img.zip",
+            "compression": "zip"
+        },
+        {
+            "filename": "chr-6.49.7.img",
+            "version": "6.49.6",
```
Comment (grossmj) [inline appliances/mikrotik-ccr1036-8g-2s+.gns3a:89]: Should be 6.49.7

File: appliances/mikrotik-ccr1072-1g-8s+.gns3a [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            "version": "7.1.5",
+            "md5sum": "9c0be05f891df2b1400bdae5e719898e",
+            "filesize": 134217728,
+            "download_url": "http://www.mikrotik.com/download",
+            "direct_download_url": "https://download.mikrotik.com/routeros/7.1.5/chr-7.1.5.img.zip",
+            "compression": "zip"
+        },
+        {
+            "filename": "chr-6.49.7.img",
+            "version": "6.49.6",
```
Comment (grossmj) [inline appliances/mikrotik-ccr1072-1g-8s+.gns3a:85]: Should be 6.49.7

File: appliances/mikrotik-crs328-24p-4s+.gns3a [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            "version": "7.1.5",
+            "md5sum": "9c0be05f891df2b1400bdae5e719898e",
+            "filesize": 134217728,
+            "download_url": "http://www.mikrotik.com/download",
+            "direct_download_url": "https://download.mikrotik.com/routeros/7.1.5/chr-7.1.5.img.zip",
+            "compression": "zip"
+        },
+        {
+            "filename": "chr-6.49.7.img",
+            "version": "6.49.6",
```
Comment (grossmj) [inline appliances/mikrotik-crs328-24p-4s+.gns3a:97]: Should be 6.49.7

File: appliances/mikrotik-crs328-4c-20s-4s+.gns3a [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+            "version": "7.1.5",
+            "md5sum": "9c0be05f891df2b1400bdae5e719898e",
+            "filesize": 134217728,
+            "download_url": "http://www.mikrotik.com/download",
+            "direct_download_url": "https://download.mikrotik.com/routeros/7.1.5/chr-7.1.5.img.zip",
+            "compression": "zip"
+        },
+        {
+            "filename": "chr-6.49.7.img",
+            "version": "6.49.6",
```
Comment (grossmj) [inline appliances/mikrotik-crs328-4c-20s-4s+.gns3a:113]: Should be 6.49.7

File: appliances/mikrotik-rb1100ahx4-dude-edition.gns3a [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+            "version": "7.1.5",
+            "md5sum": "9c0be05f891df2b1400bdae5e719898e",
+            "filesize": 134217728,
+            "download_url": "http://www.mikrotik.com/download",
+            "direct_download_url": "https://download.mikrotik.com/routeros/7.1.5/chr-7.1.5.img.zip",
+            "compression": "zip"
+        },
+        {
+            "filename": "chr-6.49.7.img",
+            "version": "6.49.6",
```
Comment (grossmj) [inline appliances/mikrotik-rb1100ahx4-dude-edition.gns3a:79]: Should be 6.49.7

File: appliances/mikrotik-rb2011uias.gns3a [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+            "version": "7.1.5",
+            "md5sum": "9c0be05f891df2b1400bdae5e719898e",
+            "filesize": 134217728,
+            "download_url": "http://www.mikrotik.com/download",
+            "direct_download_url": "https://download.mikrotik.com/routeros/7.1.5/chr-7.1.5.img.zip",
+            "compression": "zip"
+        },
+        {
+            "filename": "chr-6.49.7.img",
+            "version": "6.49.6",
```
Comment (grossmj) [inline appliances/mikrotik-rb2011uias.gns3a:85]: Should be 6.49.7

File: appliances/mikrotik-rb3011uias.gns3a [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+            "version": "7.1.5",
+            "md5sum": "9c0be05f891df2b1400bdae5e719898e",
+            "filesize": 134217728,
+            "download_url": "http://www.mikrotik.com/download",
+            "direct_download_url": "https://download.mikrotik.com/routeros/7.1.5/chr-7.1.5.img.zip",
+            "compression": "zip"
+        },
+        {
+            "filename": "chr-6.49.7.img",
+            "version": "6.49.6",
```
Comment (grossmj) [inline appliances/mikrotik-rb3011uias.gns3a:85]: Should be 6.49.7

File: appliances/mikrotik-rb4011igs+.gns3a [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+            "version": "7.1.5",
+            "md5sum": "9c0be05f891df2b1400bdae5e719898e",
+            "filesize": 134217728,
+            "download_url": "http://www.mikrotik.com/download",
+            "direct_download_url": "https://download.mikrotik.com/routeros/7.1.5/chr-7.1.5.img.zip",
+            "compression": "zip"
+        },
+        {
+            "filename": "chr-6.49.7.img",
+            "version": "6.49.6",
```
Comment (grossmj) [inline appliances/mikrotik-rb4011igs+.gns3a:85]: Should be 6.49.7

File: appliances/mikrotik-rb450g.gns3a [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+            "version": "7.1.5",
+            "md5sum": "9c0be05f891df2b1400bdae5e719898e",
+            "filesize": 134217728,
+            "download_url": "http://www.mikrotik.com/download",
+            "direct_download_url": "https://download.mikrotik.com/routeros/7.1.5/chr-7.1.5.img.zip",
+            "compression": "zip"
+        },
+        {
+            "filename": "chr-6.49.7.img",
+            "version": "6.49.6",
```
Comment (grossmj) [inline appliances/mikrotik-rb450g.gns3a:79]: Should be 6.49.7

File: appliances/mikrotik-rb450gx4.gns3a [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+            "version": "7.1.5",
+            "md5sum": "9c0be05f891df2b1400bdae5e719898e",
+            "filesize": 134217728,
+            "download_url": "http://www.mikrotik.com/download",
+            "direct_download_url": "https://download.mikrotik.com/routeros/7.1.5/chr-7.1.5.img.zip",
+            "compression": "zip"
+        },
+        {
+            "filename": "chr-6.49.7.img",
+            "version": "6.49.6",
```
Comment (grossmj) [inline appliances/mikrotik-rb450gx4.gns3a:79]: Should be 6.49.7

**Review by grossmj — CHANGES_REQUESTED** [review]
Thanks for your contribution. After a quick review, I have noticed a version mismatch in the files.

**Comment by AzorianMatt** [comment]
@grossmj you're welcome and my apologies for wasting your time on something so obvious! I'll get some fixes pushed today for this. I'm testing a method with the x86 image to see if it's a more suitable fit for a few of these.

Thanks for the fast feedback!

**Comment by grossmj** [comment]
> @grossmj you're welcome and my apologies for wasting your time on something so obvious! 

No worries at all.

Just one last note, the following appliances can be downgraded to registry version 5 because they don't use custom adapters: https://github.com/GNS3/gns3-registry#versioning

* mikrotik-rb1100ahx4-dude-edition.gns3a
* mikrotik-rb450gx4.gns3a
* mikrotik-rb450g.gns3a

Thanks! :)

**Comment by AzorianMatt** [comment]
All handled, thanks @grossmj!

**Review by grossmj — APPROVED** [review]

**Pull request merged**